analyzers/src/SonarAnalyzer.Common/Rules/Utilities/UtilityAnalyzerBase.cs

```diff
@@ -87,12 +87,20 @@ public abstract class UtilityAnalyzerBase<TMessage> : UtilityAnalyzerBase
             context.RegisterCompilationAction(c =>
                 {
                     ReadParameters(context, c);
-                    if (!IsAnalyzerEnabled ||
-                        (SkipAnalysisForTestProject && IsTestProject))
+                    if (!IsAnalyzerEnabled)
                     {
                         return;
                     }
 
+                    // The results of Metrics and CopyPasteToken analyzers are not needed for Test projects yet the plugin side expects the protobuf files, so we create empty ones.
+                    if (SkipAnalysisForTestProject && IsTestProject)
+                    {
+                        // Make sure the folder exists
```

Comment:
These 3 lines are here two times and they are not trivial. Let's move them to a method/local function to get the stream. This one will just dispose it, the other one will feed it.